### PR TITLE
<TextInput>

### PR DESCRIPTION
**Pull request by ryotogashi**

### CODE REVIEW

**Comment by ryotogashi** [comment]
@axross Can you check?

File: components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+import styled, { FlattenSimpleInterpolation, css } from "styled-components";
+
+interface Props extends React.Attributes {
+  multiline?: boolean;
+  rows?: number;
```
Comment (axross) [inline components/TextInput/TextInput.tsx]: Seems `multiline` and `rows` intersects each other. `multiline=true === rows > 1`.

So how about removing `multiline` and deciding if it's multiline or not from `rows > 1`. Then the default value of `rows` should be `1`.

File: components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  placeholder?: string;
+  size?: TextInputSize;
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => void;
```
Comment (axross) [inline components/TextInput/TextInput.tsx]:
```suggestion
  onChange?: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
```

File: components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+const TextAreaRoot = styled.textarea<{
+  size: TextInputSize;
+}>`
+  background-color: #fff;
+  border: solid 1px #576574;
+  border-radius: 4px;
+  color: #576574;
+  font-size: 16px;
+  padding-left: 4px;
```
Comment (axross) [inline components/TextInput/TextInput.tsx]: Probably you also need to care of more properties, such as `outline` and `font-family`. Look at the browser's default styles.

File: components/TextInput/TextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  [TextInputSize.large]: css`
+    width: 320px;
+  `,
+  [TextInputSize.regular]: css`
+    width: 160px;
+  `,
+  [TextInputSize.small]: css`
+    width: 80px;
+  `
+};
```
Comment (axross) [inline components/TextInput/TextInput.tsx]: Probably you didn't cover that there's "large" text input UI in the rough screen image.

Screen Shot 2020-03-16 at 8 12 07 AM

File: components/TextInput/TextInput.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import * as React from "react";
+import TextInput, { TextInputSize } from "./TextInput";
+
+export default {
+  title: "TextInput",
+  component: TextInput
+};
+
+export const example = () => (
+  <TextInput placeholder="text input" onChange={action("inputed")} />
```
Comment (axross) [inline components/TextInput/TextInput.stories.tsx]: Use `addon-knobs`'s `text`, `select`, and things like that to make all props customizable. See [`<Button>`](https://github.com/agnosticful/chipstackoverflow-web/blob/master/components/Button/Button.stories.tsx)

File: components/TextInput/TextInput.tsx (hunk 4 above)
Comment (ryotogashi) [reply components/TextInput/TextInput.tsx]: mmm, I covered with fullwidth for that

**Comment by axross** [comment]
As mentioned on LINE, I'll take it.

**Pull request closed without merging**